src/mem3/src/mem3_shards.erl

```diff
@@ -249,6 +269,21 @@ code_change(_OldVsn, #st{}=St, _Extra) ->
 
 %% internal functions
 
+start_changes_listener(SinceSeq) ->
```

Comment:
Exactly right. @jaydoane noticed that if mem3_shards restarts we can end up with two changes listeners sending updates to mem3_shards. This setup ensures that the old one dies which is important now that we're tracking state from the listener.